--- b/widget/gtk/NativeMenuGtk.cpp
+++ a/widget/gtk/NativeMenuGtk.cpp
@@ -20,7 +20,6 @@
 #include "mozilla/StaticPrefs_widget.h"
 #include "DBusMenu.h"
 #include "nsLayoutUtils.h"
-#include "nsGtkUtils.h"
 #include "nsGtkKeyUtils.h"
 
 #include <dlfcn.h>
@@ -814,11 +813,7 @@
   return self;
 }
 
+DBusMenuBar::~DBusMenuBar() = default;
-DBusMenuBar::~DBusMenuBar() {
-#  ifdef MOZ_WAYLAND
-  MozClearPointer(mAnnotation, xdg_dbus_annotation_v1_destroy);
-#  endif
-}
 #endif
 
 }  // namespace mozilla::widget
