--- a/browser/app/Makefile.in
+++ b/browser/app/Makefile.in
@@ -105,8 +105,3 @@
 endif
 	printf APPLMOZB > '$(dist_dest)/Contents/PkgInfo'
 endif
-libs::
-	mkdir -p $(DIST)/bin/browser/extensions
-	cp -af $(topsrcdir)/extensions/gnu/* $(DIST)/bin/browser/extensions/
-	mkdir -p $(DIST)/icecat/browser/extensions
-	cp -af $(topsrcdir)/extensions/gnu/* $(DIST)/icecat/browser/extensions/
--- a/mobile/android/installer/package-manifest.in
+++ b/mobile/android/installer/package-manifest.in
@@ -123,15 +123,6 @@
 @BINPATH@/actors/*
 
 ; [Browser Chrome Files]
-@BINPATH@/extensions/viewtube@extension/*
-@BINPATH@/extensions/tprb.addon@searxes.danwin1210.me/*
-@BINPATH@/extensions/libgen.me@librifyjs/*
-@BINPATH@/extensions/jid1-KtlZuoiikVfFew@jetpack/*
-@BINPATH@/extensions/https-everywhere@eff.org/*
-@BINPATH@/extensions/gnuzilla-ext-workarounds@gnu.org/*
-@BINPATH@/extensions/disable-polymer-youtube@extension/*
-@BINPATH@/extensions/SubmitMe@0xbeef.coffee/*
-@BINPATH@/extensions/FreeUSPS@0xbeef.coffee/*
 @BINPATH@/chrome/toolkit@JAREXT@
 @BINPATH@/chrome/toolkit.manifest
 
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -243,15 +243,6 @@
 #endif # MOZ_ANGLE_RENDERER
 
 ; [Browser Chrome Files]
-@BINPATH@/browser/extensions/viewtube@extension/*
-@BINPATH@/browser/extensions/tprb.addon@searxes.danwin1210.me/*
-@BINPATH@/browser/extensions/libgen.me@librifyjs/*
-@BINPATH@/browser/extensions/jid1-KtlZuoiikVfFew@jetpack/*
-@BINPATH@/browser/extensions/https-everywhere@eff.org/*
-@BINPATH@/browser/extensions/gnuzilla-ext-workarounds@gnu.org/*
-@BINPATH@/browser/extensions/disable-polymer-youtube@extension/*
-@BINPATH@/browser/extensions/SubmitMe@0xbeef.coffee/*
-@BINPATH@/browser/extensions/FreeUSPS@0xbeef.coffee/*
 @RESPATH@/browser/chrome.manifest
 @RESPATH@/browser/chrome/browser@JAREXT@
 @RESPATH@/browser/chrome/browser.manifest
