From c3a7cccbef535051541dac431e0863f1b203779b Mon Sep 17 00:00:00 2001
From: Chippy <chippy@classictetris.net>
Date: Tue, 12 Sep 2023 11:18:36 +0200
Subject: [PATCH 08/13] whole part removed aupstream and renamed two files also
 upstream

---
 makeicecat | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/makeicecat b/makeicecat
index b792a55..0fd893a 100755
--- a/makeicecat
+++ b/makeicecat
@@ -442,7 +442,8 @@ EOF
 configure_search()
 {
     # Leak less info to DuckDuckGo.
-    sed '/"params"/,/],/d' -i browser/components/search/extensions/ddg/manifest.json
+    # Chippy 20230912 this has been removed upstream
+    #sed '/"params"/,/],/d' -i browser/components/search/extensions/ddg/manifest.json
 
     # Default DDG to HTML search.
     cp -a browser/components/search/extensions/ddg browser/components/search/extensions/ddg-js
@@ -633,8 +634,8 @@ s/OpenSource/Free Software/g;
     find . -name region.properties | xargs -i /bin/sed 's_https://www\.mibbit.*__' -i {}
 
     # Set migrator scripts
-    cp browser/components/migration/IceCatProfileMigrator.jsm browser/components/migration/FirefoxProfileMigrator.jsm
-    sed 's/IceCat/Firefox/g; s/icecat/firefox/g' -i browser/components/migration/FirefoxProfileMigrator.jsm
+    cp browser/components/migration/IceCatProfileMigrator.jsm browser/components/migration/FirefoxProfileMigrator.sys.mjs
+    sed 's/IceCat/Firefox/g; s/icecat/firefox/g' -i browser/components/migration/FirefoxProfileMigrator.sys.mjs
 
     # Copy js settings
     cat "${DATADIR}"/settings.js >> browser/app/profile/icecat.js
-- 
2.40.1

