From 2ddadbf64a07b6f60624be7de7d11a55d0561950 Mon Sep 17 00:00:00 2001
From: Chippy <chippy@classictetris.net>
Date: Tue, 12 Sep 2023 16:15:47 +0200
Subject: [PATCH 10/13] slight change upstream

---
 makeicecat | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makeicecat b/makeicecat
index f524130..ca12b0b 100755
--- a/makeicecat
+++ b/makeicecat
@@ -624,7 +624,7 @@ s/OpenSource/Free Software/g;
     find . -depth | grep run-mozilla | rename_files -e s/mozilla/icecat/g
 
     # do not alter useragent/platform/oscpu/etc with fingerprinting countermeasure, it makes things worse
-    sed '/ShouldResistFingerprinting/,/}/s/^/\/\//' -i ./netwerk/protocol/http/nsHttpHandler.cpp
+    sed '/if (aShouldResistFingerprinting/,/}/s/^/\/\//' -i ./netwerk/protocol/http/nsHttpHandler.cpp
     sed '/If fingerprinting resistance is on/,/}/s/^/\/\//' -i ./dom/base/Navigator.cpp
 
     # Leave user agent as Firefox
-- 
2.40.1

