From dc95e32e4efcd06f46719ec10c78c78626dd6c9c Mon Sep 17 00:00:00 2001
From: Chippy <chippy@classictetris.net>
Date: Tue, 12 Sep 2023 21:50:18 +0200
Subject: [PATCH 11/13] untested change with tac and version bump

---
 makeicecat | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/makeicecat b/makeicecat
index ca12b0b..28d7ba0 100755
--- a/makeicecat
+++ b/makeicecat
@@ -22,10 +22,10 @@
 
 # metadata
 readonly FFMAJOR=115
-readonly FFMINOR=1
-readonly FFSUB=0
+readonly FFMINOR=2
+readonly FFSUB=1
 readonly GNUVERSION=1
-readonly SOURCEBALL_CHECKSUM='a7c92e576b678854c0ea83062e7d0c97936947fca30865129f47734e806e17a6'
+readonly SOURCEBALL_CHECKSUM='5439b5b603ad7e60e85723a497f7ff761ad0a2cc8a685ffb5a1c269b1250c966'
 readonly SOURCEBALL_SIGNINGKEY='14F26682D0916CDD81E37B6D61B7B526D98F0353'
 readonly FFVERSION=${FFMAJOR}.${FFMINOR}.${FFSUB}
 readonly SOURCEBALL=firefox-${FFVERSION}esr.source.tar.xz
@@ -526,8 +526,9 @@ apply_branding()
     # Delete stuff we don't use and that may contain trademarked logos
     rm -rf ./browser/metro ./addon-sdk/source/doc/static-files/media ./b2g
 
-    # Custom bookmarks
-    sed '/ifndef/,/endif/d' -i browser/base/content/default-bookmarks.html
+    # Custom bookmarks the content of the target file has changed and now endif is present more than once
+    #sed '/ifndef/,/endif/d' -i browser/base/content/default-bookmarks.html
+    tac browser/base/content/default-bookmarks.html | sed '/endif/,/ifndef/d' | tac > tmpfile > browser/base/content/default-bookmarks.html && rm tmpfile
 
     # Custom legal about pages
     find l10n -wholename '*/brand.dtd' | xargs /bin/sed 's/trademarkInfo.part1.*/trademarkInfo.part1 "The IceCat logo is Copyright 2008-2015 Free Software Foundation, released under the terms of the GNU Lesser General Public License, version 3 or any later version.">/' -i
-- 
2.40.1

