diff --git a/browser/moz.configure b/browser/moz.configure
index e8b401a7dfb2..d5b3117ce3d8 100644
--- a/browser/moz.configure
+++ b/browser/moz.configure
@@ -5,16 +5,19 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 imply_option("MOZ_PLACES", True)
-imply_option("MOZ_SERVICES_HEALTHREPORT", True)
-imply_option("MOZ_SERVICES_SYNC", True)
-imply_option("MOZ_DEDICATED_PROFILES", True)
-imply_option("MOZ_BLOCK_PROFILE_DOWNGRADE", True)
-imply_option("MOZ_NORMANDY", True)
+imply_option("MOZ_SERVICES_HEALTHREPORT", False)
+ # Disabling MOZ_SERVICES_SYNC would break runtime javascript,
+ # see <https://bugzilla.mozilla.org/show_bug.cgi?id=1816969>.
+imply_option("MOZ_SERVICES_SYNC", True)
+imply_option("MOZ_DEDICATED_PROFILES", False)
+imply_option("MOZ_BLOCK_PROFILE_DOWNGRADE", False)
+imply_option("MOZ_NORMANDY", False)
 imply_option("MOZ_PROFILE_MIGRATOR", True)
 
 
-imply_option("MOZ_APP_VENDOR", "Mozilla")
+imply_option("MOZ_APP_VENDOR", "GNU")
 imply_option("MOZ_APP_ID", "{ec8030f7-c20a-464f-9b0e-13a3a9e97384}")
+imply_option("MOZ_APP_PROFILE", "mozilla/firefox")
 # Include the DevTools client, not just the server (which is the default)
 imply_option("MOZ_DEVTOOLS", "all")
 imply_option("BROWSER_CHROME_URL", "chrome://browser/content/browser.xhtml")
diff --git a/build/moz.configure/init.configure b/build/moz.configure/init.configure
index 193272588caa..da8a729548c8 100644
--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -991,7 +991,7 @@ set_config("MOZ_BUILD_APP", build_project)
 set_define("MOZ_BUILD_APP", build_project)
 
 
-option(env="MOZILLA_OFFICIAL", help="Build an official release")
+option(env="MOZILLA_OFFICIAL", help="Build an official release", default=False)
 
 
 @depends("MOZILLA_OFFICIAL")
