--- a/examples/Makefile.am	2022-03-21 16:10:53.638799827 +0200
+++ b/examples/Makefile.am	2022-03-21 16:11:31.893804336 +0200
@@ -1,7 +1,7 @@
 # Google BSD license https://developers.google.com/google-bsd-license
 # Copyright 2012 Google Inc. wrightt@google.com
 
-AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/include/ios-webkit-debug-proxy -I$(top_srcdir)/src
 
 AM_CFLAGS = $(GLOBAL_CFLAGS) $(libimobiledevice_CFLAGS) $(libplist_CFLAGS) $(openssl_CFLAGS)
 AM_LDFLAGS = $(libimobiledevice_LIBS) $(libplist_LIBS) $(openssl_LIBS)
